src/daemon/daemon.go

```diff
@@ -1081,6 +1083,16 @@ func (dm *Daemon) handleMessageSendResult(r gnet.SendResult) {
 	switch r.Message.(type) {
 	case SendingTxnsMessage:
 		dm.announcedTxns.add(r.Message.(SendingTxnsMessage).GetFiltered())
+	case DisconnectPeerMessage:
```

Comment:
@gz-c what do you think about this approach for synchronizing disconnection after `RejectMessage` is successfully sent? It's relatively simple . I plan to submit another patch for #1957 .

Reply:
Open question : Should disconnection happen only on success or also on error

Reply:
I think Disconnect should always be called even if there was an error attached to the message.

This approach probably works, but you'd want to call disconnect directly on the pool, and change daemon.Disconnect to send a disconnect packet instead.